src/common/perf_counters_cache.cc

```diff
@@ -0,0 +1,116 @@
+#include "common/perf_counters_cache.h"
+#include "common/perf_counters_key.h"
+
+namespace ceph::perf_counters {
+
+void PerfCountersCache::check_key(const std::string &key) {
+  std::string_view key_name = ceph::perf_counters::key_name(key);
+  // return false for empty key name
+  assert(key_name != "");
+
+  // if there are no labels key name is not valid
+  auto key_labels = ceph::perf_counters::key_labels(key);
+  assert(key_labels.begin() != key_labels.end());
+
+  // don't accept keys where any labels have an empty label name
+  for (auto key_label : key_labels) {
+    assert(key_label.first != "");
+    assert(key_label.second != "");
```

Comment:
as before, we should only require the label *key* to be non-empty. please remove this assert on the label value (`.second`)